sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/PlanUtils.java

```diff
@@ -76,8 +77,114 @@ public static boolean assetConflicts(PodInstanceRequirement asset, Collection<Po
+            return false;
+        }
+        if (element instanceof Step) {
+            Optional<PodInstanceRequirement> podInstanceRequirement = ((Step) element).getPodInstanceRequirement();
+            if (podInstanceRequirement.isPresent()
+                    && PlanUtils.assetConflicts(podInstanceRequirement.get(), dirtyAssets)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Returns the overall status to display by a parent element of the provided children.
+     *
+     * @param parentName the name of the parent element, used for logging
+     * @param childStatuses the statuses of the parent element's children
+     * @param candidateStatuses the statuses of the parent element's candidate children, as selected by the parent
+     *      element's Strategy
+     * @param errors any errors from the parent element
+     * @param isInterrupted whether the parent element is interrupted
+     * @return the Status to display for the parent element
+     */
+    public static <C extends Element> Status getParentStatus(
+            String parentName,
+            Collection<Status> childStatuses,
+            Collection<Status> candidateStatuses,
+            Collection<String> errors,
+            boolean isInterrupted) {
+        // Ordering matters throughout this method.  Modify with care.
+        // Also note that this function MUST NOT call parent.getStatus() as that creates a circular call.
+        Status result;
+
+        if (!errors.isEmpty() || anyMatch(Status.ERROR, childStatuses)) {
+            result = Status.ERROR;
+            LOGGER.debug("({} status={}) One or more children contain errors.", parentName, result);
+        } else if (allMatch(Status.COMPLETE, childStatuses)) {
+            result = Status.COMPLETE;
+            LOGGER.debug("({} status={}) All children have status: {}", parentName, result, Status.COMPLETE);
+        } else if (isInterrupted) {
+            result = Status.WAITING;
+            LOGGER.debug("({} status={}) Parent element is interrupted", parentName, result);
+        } else if (anyMatch(Status.PREPARED, childStatuses)) {
+            result = Status.IN_PROGRESS;
+            LOGGER.debug("({} status={}) At least one child has status: {}", parentName, result, Status.PREPARED);
+        } else if (anyMatch(Status.WAITING, candidateStatuses)) {
+            result = Status.WAITING;
+            LOGGER.debug("({} status={}) At least one candidate has status: {}", parentName, result, Status.WAITING);
+        } else if (anyMatch(Status.IN_PROGRESS, candidateStatuses)) {
+            result = Status.IN_PROGRESS;
+            LOGGER.debug("({} status={}) At least one candidate has status: {}",
+                    parentName, result, Status.IN_PROGRESS);
+        } else if (anyMatch(Status.COMPLETE, childStatuses) && anyMatch(Status.PENDING, candidateStatuses)) {
```

Comment:
Could we not replace all of the `else if (anyMatch(Status.COMPLETE, childStatuses) && ` conditionals here with:
```java
} else if (anyMatch(Status.COMPLETE, childStatuses)) {
    result = Status.IN_PROGRESS;
}
```
(as we already know that `(allMatch(Status.COMPLETE, childStatuses))` is false at this point)

Reply:
(I could be missing something in the relationship between `childStatuses` and `candidateStatuses` here.)

Reply:
I think this is specifically checking that:
- The children contain a COMPLETE status
- The subset of children _which are candidates for doing work_ contain a PENDING status